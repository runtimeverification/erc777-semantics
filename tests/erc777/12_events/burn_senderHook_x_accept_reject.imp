test.preamble();

// burn(),  operatorBurn() through default operator (unless otherwise specified), operatorBurn() from caller.
// 2 Holder addresses, one accepting another rejecting.
//  Burn by one operator from 1st, then 2nd, then 1st.
// Hook addresses 1 and 2 are accepting.

test.registerTokensToSendHook(5, 1); //accepting
test.registerTokensToSendHook(6, 5); //rejecting
test.makeCaller(8);
test.operatorBurn(5, 2, ""); //accept
test.operatorBurn(6, 2, ""); //revert
test.makeCaller(6);
test.burn(2);                //revert
test.makeCaller(8);
test.operatorBurn(5, 2, ""); //accept

printLogs;
