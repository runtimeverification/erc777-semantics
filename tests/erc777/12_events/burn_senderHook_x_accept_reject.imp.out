Registering holder 5 with hook 1
Registering holder 6 with hook 5
Changing the caller to 8
8 is burning 2 units from 5 (operatorData:"")
tokensToSend: Hook address = 1, Operator = 8, From = 5, To = 0, Value = 2, Data = , OperatorData = 
tokensToSend: sender balance: 50
tokensToSend: receiver balance: 0
Balance of 5 is 48
Total supply is 548
8 is burning 2 units from 6 (operatorData:"")
tokensToSend: Hook address = 5, Operator = 8, From = 6, To = 0, Value = 2, Data = , OperatorData = 
tokensToSend: sender balance: 60
tokensToSend: receiver balance: 0
Reverting...
REVERT
Balance of 6 is 60
Total supply is 548
Changing the caller to 6
Burning 2 units from 6
tokensToSend: Hook address = 5, Operator = 6, From = 6, To = 0, Value = 2, Data = , OperatorData = 
tokensToSend: sender balance: 60
tokensToSend: receiver balance: 0
Reverting...
REVERT
Balance of 6 is 60
Total supply is 548
Changing the caller to 8
8 is burning 2 units from 5 (operatorData:"")
tokensToSend: Hook address = 1, Operator = 8, From = 5, To = 0, Value = 2, Data = , OperatorData = 
tokensToSend: sender balance: 48
tokensToSend: receiver balance: 0
Balance of 5 is 46
Total supply is 546

Events:
Minted(8, 10, 100, , init)
Transfer(0, 10, 100)
Minted(8, 9, 90, , init)
Transfer(0, 9, 90)
Minted(8, 8, 80, , init)
Transfer(0, 8, 80)
Minted(8, 7, 70, , init)
Transfer(0, 7, 70)
Minted(8, 6, 60, , init)
Transfer(0, 6, 60)
Minted(8, 5, 50, , init)
Transfer(0, 5, 50)
Minted(8, 4, 40, , init)
Transfer(0, 4, 40)
Minted(8, 3, 30, , init)
Transfer(0, 3, 30)
Minted(8, 2, 20, , init)
Transfer(0, 2, 20)
Minted(8, 1, 10, , init)
Transfer(0, 1, 10)
Burned(8, 5, 2, )
Transfer(5, 0, 2)
Burned(8, 5, 2, )
Transfer(5, 0, 2)
